### ISSUE TITLE
Automatically populate First name and Last name from IDIR

### ISSUE THREAD

**Issue by chrislaick** [bcgov/MFIN-Data-Catalogue#579]
https://github.com/bcgov/MFIN-Data-Catalogue/labels/Idea

---

## Describe the idea

Explore removing the Drupal module realname. Currently the First name and Last name field of a User is set to required. I can imagine these fields are being used in emails e.g. "Hello [firstname]". 

However, on registration with IDIR, the user is **not** prompted to enter their First name or Last name, nor is this information pulled from the IDIR identity provider. Therefore, users will only fill in their name if they decide to edit their profile and enter their name.

If the First name and Last name fields are used in elsewhere in the system, these fields should always populated. We can do this by pulling in data from IDIR attributes. We may not need the realname module after all as it interferes with the Username field for a User.

In summary, there are two benefits to removing realname:

- Configuring OpenID Connect to pull in First name and Last name automatically from the IDP, so these fields are always populated.
- Exposing the Username field to allowing us power users to create our **own** Drupal accounts for testing 

## Actions

- [x] Conduct a dependency check to see where First name and Last name fields are used
- [x] Remove Real Name module
- [x] Configure OpenID Connect to pull in First name and Last name automatically
- [x] Clean up existing users so we have consistent usernames

**Comment by NicoledeGreef** [comment]
This explains why when we go in to **People** as Administrators and try to save a change (usually adding their Organization) to a new user's profile we are forced to enter First and Last Name. If we can automatically pull in the info and populate that would be an improvement.  

Similarly, we were hoping to assess an IDIR user's security group membership and apply their Organization value automatically where possible (see #424) but there seemed to be a gap in what attributes we could get at last year.

**Comment by chrislaick** [comment]
Deployed to DEV and TEST and ready for QA. One way to test this is to delete an IDIR account and login again. The first and last name fields should be automatically populated. We can now easily create Drupal accounts for our testing purposes.

**Comment by NicoledeGreef** [comment]
@david-fong-bc : let's collaborate on testing as Chris suggested in the previous comment (delete one of our accounts in Test then try logging in and check the auto-setup and ensure our roles are manually re-applied). I'll set up a time for us to do this test.

Another test I did solo:

- In Test, I added a user "De Greef Test" which was given a random string as an ID:
![Image](https://github.com/user-attachments/assets/fd36e54c-844e-445d-ada1-2b0a81489174)
- I gave the account some roles
- Only when I visit the Edit profile screen do I see the First and Last Name I supplied.
![Image](https://github.com/user-attachments/assets/7660d239-743f-4ed3-9900-693713b1d9cb)
- In the user list I cannot filter to find the account by searching for the first or last name values. If search for "test" only one account is returned:
![Image](https://github.com/user-attachments/assets/e154c8eb-e962-44ee-82f2-bdd227b0c70a)

My question to @chrislaick is: how would I login as that user if there is no email and password to supply at https://test.cat.data.fin.gov.bc.ca/user/login?showcore

**Comment by NicoledeGreef** [comment]
Collaborative testing:

Attempted to:  Cancel David's account and leave content with _Anonymous_ 
This resulted in a Gateway timeout error:

An AJAX HTTP error occurred.
HTTP Result Code: 504
Debugging information follows.
Path: /batch?id=157&op=do_nojs&op=do
StatusText: Gateway Time-out
ResponseText: 504 Gateway Time-out
The server didn't respond in time.

`45 items successfully processed:<div class="item-list"><ul><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/property-transfer-tax-ptt-return" hreflang="en">Property Transfer Tax (PTT) return</a></li><li><a href="/david-test-2025-03-25-data-1" hreflang="en">david test 2025-03-25 data</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data" hreflang="en">david test 2025-03-25 data</a></li><li><a href="/david-test-2025-03-25-data-0" hreflang="en">david test 2025-03-25 data</a></li><li><a href="/david-test-2025-03-25-data-0" hreflang="en">david test 2025-03-25 data</a></li><li><a href="/david-test-2025-03-25-data-0" hreflang="en">david test 2025-03-25 data</a></li><li><a href="/iceacdqueue" hreflang="en">ICE_ACDQueue</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li><li><a href="/david-test-2025-03-25-data-2" hreflang="en">david test 2025-03-25 data 2</a></li></ul></div>`

**Comment by NicoledeGreef** [comment]
Cancellation form: add text something along the lines of: to delete account please use the bulk owner change prior to deleting account.

**Comment by chrislaick** [comment]
Make all new IDIR registrations assign Data Catalogue user as a role.

**Comment by danhgov** [comment]
@chrislaick, I pulled your commits on this ticket out of the main branch and into their own branch, so that I could deploy everything else. See [issue-579](https://github.com/bcgov/MFIN-Data-Catalogue/tree/issue-579 ).

**Comment by chrislaick** [comment]
After sitting on this for a while, I've decided to keep realname in the codebase as there are more changes needed to fix the username with auto_username a required module for bc_dc. Instead I've just configured openid_connect to automatically pull in:

- Given name -> First name
- Family name -> Last name

This means these fields will get automatically populated, but users have the flexibility of changing these values in 'Edit Profile' if they want to. Lastly, if there is a need to create Drupal test accounts, we, administrators can do that on an ad-hoc basis.

**Comment by chrislaick** [comment]
Deployed to DEV and TEST. Marking Ready for QA.

**Comment by NicoledeGreef** [comment]
In **Dev**, I verified this item by removing First and Last name values from my profile and saving. Then using Incognito to force a fresh login, I visited my profile in Dev again and saw that the values had been updated correctly. 

When I tried the same thing in **Test**, however, I got an error upon my attempt to save after removing my first and last name values:

![Image](https://github.com/user-attachments/assets/b1cff276-90a6-421e-88ca-48332301b891)

**Comment by chrislaick** [comment]
@NicoledeGreef Thanks for testing this. This happened because OP Admin (User 1) in TEST was set to your email address. I've removed your email address from OP Admin and secured/blocked User 1. You may try the test again and it should work this time.

**Comment by NicoledeGreef** [comment]
Thanks @chrislaick  - verified in Test now as well.  I figured there might be an easy explanation for the difference. Thanks for looking into it!

**Comment by chrislaick** [comment]
Deployed to PROD. Marking as Done.
